src/libraries/Common/src/System/Security/Cryptography/MLKem.cs

```diff
@@ -1801,32 +1815,26 @@ private TResult ExportPkcs8PrivateKeyCallback<TResult>(ExportPkcs8PrivateKeyFunc
             // Decapsulation keys are always larger than the seed, so if we end up with a seed export it should
             // fit in the initial buffer.
             int size = Algorithm.DecapsulationKeySizeInBytes + 32;
-            byte[] buffer = ArrayPool<byte>.Shared.Rent(size); // Released to callers, do not use CryptoPool.
+            byte[] buffer = CryptoPool.Rent(size); // Only passed out as span, callees can't keep a reference to it
             int written;
 
             while (!TryExportPkcs8PrivateKeyCore(buffer, out written))
             {
-                ClearAndReturnToPool(buffer, written);
+                CryptoPool.Return(buffer);
                 size = checked(size * 2);
                 buffer = ArrayPool<byte>.Shared.Rent(size);
             }
 
-            if (written > buffer.Length)
+            if (written < 0 || written > buffer.Length)
             {
                 // We got a nonsense value written back. Clear the buffer, but don't put it back in the pool.
                 CryptographicOperations.ZeroMemory(buffer);
                 throw new CryptographicException();
             }
 
             TResult result = func(buffer.AsSpan(0, written));
```

Comment:
Should this invocation be guarded by a `try..catch`?

Reply:
`func` is going to either be

1. `static pkcs8 => PemEncoding.WriteString(PemLabels.Pkcs8PrivateKey, pkcs8)`
2. `static pkcs8 => pkcs8.ToArray()`

I don't really expect writing a PEM or `ToArray` to ever throw.